### DISCUSSION TITLE
OAuth / Realtime API not working when behind a reverse proxy in a container (Coolify)

### DISCUSSION THREAD

**Discussion by mistuttodos** [pocketbase/pocketbase#5593] [Q&A] [answered]
I first want to thank you for this amazing product! Works flawlessly, but I have an issue in a specific use case.

I host a Coolify instance on a Hetzner VPS that uses Traefik and automatically manages Pocketbase instances.
Each Pocketbase instance has a subdomain assigned.

From Sveltekit, when I try to login with OAuth, it opens a blank page and it will soon timeout and come with:

`ClientResponseError 0: Something went wrong while processing your request.`

and e.originalError:

`Error: EventSource connect took too long.`

Running latest version of Pocketbase and Pocketbase-JS.
There is no firewall configured on Hetzner.

PS: When I use the exact same configuration using a pockethost instance URL or a local one, it works just fine, so I assume the issue is related to how the traffic is forwarded.

Edit: Accessing https://x.y.pocketbase.url/api/realtime from the browser hangs for a very long time.

**Comment by mistuttodos** [comment]
Found a solution to get it to work. (At least for now) :)

Maybe it will help someone in the future.

Seems like the experimental Caddy support for Coolify is actually working as expected and after some testing I haven't found any problem with it.

Realtime API also works fine.

So for anyone going through the same frustration, just go to your coolify instance (make sure to use the instance IP, not the hostname, as the proxy will be off for a while) -> Servers -> Localhost -> Proxy -> Turn it OFF -> Press 'Switch Proxy' that is now available -> Choose Caddy -> DONE!

Wait for some time for the certs to get generated and you're good to go! 👍

**Comment by orenaksakal** [comment]
hey, I maintain the coolify pocketbase template; traefik would work as well just make sure you disable gzip compression which is enabled by default in deployment settings

**Reply by mistuttodos** [reply]
Hi, thanks for the reply!

I already tried to do that, but didn't manage to get it to work fine (edited the config in the /data/coolify directory and set http(s).compress=false).

Also, given that Caddy also has (from what I saw in the config) gzip compression enabled, how is it that it works? Is Traefik more selective/granular with header forwarding compared to Caddy?

**Reply by orenaksakal** [reply, marked as answer]
have you tried this?
![Screenshot 2024-10-03 at 8 05 19 PM](https://github.com/user-attachments/assets/aaec9d96-8bf7-454a-8f35-0568bef91289)

I'm not sure about the traefik vs caddy gzip differences but you can't compress SSEs thats the main problem

**Reply by mistuttodos** [reply]
No, just went back to it to see there is a config on the individual services... 

I'll open a Coolify issue tomorrow to maybe improve the layout there or maybe add a note about it in the docs related to services, been using it for a few weeks, but never saw that button 💧 Probably would have saved many hours

THANKS! This should be marked as the answer for the thread. Excited to see what other services' settings have to offer

**Reply by levino** [reply]
This also brought me some additional grey hairs. It would be nice if the compression feature could be (permanently) disabled for pocket base.

**Reply by manegame** [reply]
@orenaksakal you savior. !
Always nice to realise your knowledge of the matter is just too rudimentary. Live and learn. Thanks for sharing

**Comment by arraylabs** [comment]
Thank you for the gzip hint, was going crazy until i found here to disable that.

**Comment by andrasbacsai** [comment]
I will change it to be off by default for Pocketbase (Coolify v420+). So you don't need to take care of this.

**Reply by levino** [reply]
If I may ask: What does this mean technically? Could you link a commit with the changes please? Thank you.

**Reply by dustinknopoff** [reply]
<img width="1295" height="455" alt="Screenshot 2025-09-05 at 8 33 30 PM" src="https://github.com/user-attachments/assets/ca44b428-7f9c-458b-a41e-c47947b3efb3" />

Hi @andrasbacsai I'd bumping into this issue on my coolify instance. Previously oauth was working but since upgrading to v420x it no longer does. Is it possible that this is related? (The checkbox is disabled but checked)

**Comment by manegame** [comment]
Amazing

Manus Nijhoff

0 6 - 1 8 9 7 5 5 4 8
